docs/proposals/tracing.md

```diff
@@ -0,0 +1,138 @@
+# Tracing
+
+## Context
+
+There are many tracing technologies in the market. To use tracing in your application you must instrument it with a library, therefore changing underlying technology is expensive (you have to change it in every app).
+That's why OpenTracing was created. Instrument your application once and change the backend (tracer) when you want. Around the same time OpenCensus was created which had a same goal.
+OpenCensus and OpenTracing have merged into CNCF OpenTelemetry, but the work is still in progress.
+
+Envoy has a [built-in support](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/tracing) for tracing and we want to leverage this support in Kuma.
+
+### Supported backends
+
+Envoy supports following backends: Lightstep, Zipkin,  OpenTracing (with dynamically loaded library), Datadog, OpenCensus (including GCP Stackdriver) and gRPC service for custom implementation.
+
+### Tracing as a Bootstrap Configuration
+
+You can define only one tracing backend for an Envoy and this setting is a part of the bootstrap configuration (the one that CP delivers before the start of DP).
+That means that if you want to change it, you have to restart Envoy.
+
+Also if a user deploy Kuma first and then wants to enable tracing, they need to restart Dataplanes.
+This is definitely not ideal UX, but it can probably be solved with Envoy's hot restart.
+
+### Application instrumentation
+
+Contrary to common believes, tracing is not drop-in, transparent solution. You still have to instrument your application,
+because Envoy has to correlate request that is coming in to Envoy and the App and request that is coming out from the App and Envoy.
+
+What needs to be done from an application perspective is to pass the standard [`x-request-id`](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-request-id) Envoy header. This way, Envoy can correlate requests and pass tracing headers.
+If you want to leverage other backend features like Baggage from OpenTracing you need to use library in your application and pass headers that given backend can understand (for example B3 for Zipkin).  
+
+Additionally, in my opinion the most value from tracing comes from having trace-id in application logs. This means that the application will have to read tracing header anyway and use it in logs. 
+
+The real value from introducing tracing via service mesh is to offload applications from the task of sending traces to the backend of your choice as well as being able to upgrade (or change if you use OpenTracing/OpenCensus) the tracing backend flexibly.
+
+## Requirements
+
+Kuma Dataplane can:
+* Deliver traces to the backend of your choice (one chosen backend for the first implementation)
+* Generate traces if it did not received any
```

Comment:
What does it mean ?

Reply:
If an Envoy did not received any tracing headers it will generate those and pass it to the next service.